### ISSUE TITLE
Does not work with Zabbix 3

### ISSUE THREAD

**Issue by icovada** [RussianFox/imap#31]
This does not work with Zabbix 3.

Most I managed to do is display the map and all the hosts but without getting trigger data so everything displays green

**Comment by allexxf** [comment]
Yes in zabbix 3 change API and some fields will not display. I changed the code and triggers steel display. In imap.php in block

> ```
>     if ($action_ajax=='get_triggers') {
>             $options['expandData'] = true;
>             $options['expandDescription'] = true;
>             $options['selectLastEvent'] = 'extend';
>             $options['monitored'] = true;
>             $options['maintenance'] = false;
>             $options['skipDependent'] = true;
>             $options['sortfield'] = array('lastchange');
>             $options['sortorder'] = 'DESC';
>             $options['filter'] = array('value' => TRIGGER_VALUE_TRUE);
> ```

add line

>  $options['selectHosts'] = 'hostid';

And in imap/imap.js in function loadTriggers() replace

> ```
>                               for (var nn in data) {
>                                     var trigger = data[+nn];
>                                     if (!trigger) continue;
>                                     if (!trigger.value==1) continue;
>                                     if (!_imap.markersList[trigger.hostid]) continue;
>                                     _imap.markersList[trigger.hostid].triggers[trigger.triggerid] = trigger;
> 
>                                     _imap.markersList[trigger.hostid].triggers[trigger.triggerid].lhi = lhi;
>                                     addLastTrigger(trigger);
>                                     luhost[trigger.hostid] = trigger.hostid;
>                             };
> ```

On:

> ```
>                               for (var nn in data) {
> 
>                                     var trigger = data[+nn];
>                                     if (!trigger) continue;
>                                     if (!trigger.value==1) continue;
>                                     if (!_imap.markersList[trigger.hosts[0].hostid]) continue;
>                                     _imap.markersList[trigger.hosts[0].hostid].triggers[trigger.triggerid] = trigger;
> 
>                                     _imap.markersList[trigger.hosts[0].hostid].triggers[trigger.triggerid].lhi = lhi;
>                                     addLastTrigger(trigger);
>                                     luhost[trigger.hosts[0].hostid] = trigger.hosts[0].hostid;
>                             };
> ```

This change not perfect, but work on my zabbix 3.0

**Comment by vetash** [comment]
i think there need to create new branch with these edits @RussianFox

**Comment by icovada** [comment]
Confirmed it works.

**Comment by falatronic** [comment]
@icovada You can write what changes to make to view a map?

**Comment by allexxf** [comment]
@falatronic write alex@ltd.com.ua

**Comment by tmcdon89** [comment]
I am also interested in what steps were taken to view the map on version 3.

**Comment by boyemillar** [comment]
Is there anyone working on a Zabbix v3.0 version of this

**Comment by tmcdon89** [comment]
I see it was self-assigned about a week ago. Is this something that is actively being worked on?

**Comment by brandelf68** [comment]
On zabbix 3.0 does not display the top panel filtering host groups, hosts, and trigerrs. Can anyone tell who experienced this?
In advance I thank the author for his work :)

Great work!
https://www.zabbix.com/forum/showpost.php?p=182386&postcount=428
